### PR TITLE
Related Items PR : switching over to Context API

### PR DESCRIPTION
**Pull request by Acid-Override**

### CODE REVIEW

File: client/src/components/RelatedItems/Related.jsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ function Related({ setProductId }) {
           <FaArrowAltCircleRight className="right-arrow" onClick={nextSlide} />
         ) : null}
 
-        {combinedAPIDetails.map((ea, idx) => (
+        {joinedAPIDetails.map((ea, idx) => (
           <div key={idx}>
             {/* {console.log('Testing', ea.product_id)} */}
```
Comment (addisonhernandez) [inline client/src/components/RelatedItems/Related.jsx:58]: 🔥🔥🔥🔥🔥

File: client/src/components/RelatedItems/RelatedItems.jsx [hunk 2]
```diff
@@ -47,35 +48,37 @@ function RelatedItems() {
...
+  useEffect(() => {
+    if (yourOutfitId) {
+      const getYOImages = axios.get(`/products/${yourOutfitId}/styles`);
+      const getYODetails = axios.get(`/products/${yourOutfitId}`);
+      axios
+        .all([getYOImages, getYODetails])
+        .then((response) =>
+          setYourOutfit([
+            ...yourOutfit,
+            { ...response[0].data, ...response[1].data },
```
Comment (addisonhernandez) [inline client/src/components/RelatedItems/RelatedItems.jsx:66]: 🔥 Love this. Nice use of `...` spread operator!

File: client/src/components/RelatedItems/YourOutfit.jsx [hunk 3]
```diff
@@ -1,7 +1,43 @@
...
+        {yourOutfit.length > 0 &&
+          yourOutfit.map((product, index) => (
+            <div key={index}>
+              <div className="yo-container">
+                <img
+                  className="yo-image"
+                  src={product.results[0].photos[0].thumbnail_url}
+                  alt="clothing"
+                  key={product.product_id}
+                  onClick={() => setProductId(product.product_id)}
```
Comment (addisonhernandez) [inline client/src/components/RelatedItems/YourOutfit.jsx:32]: Should the items in the Your Outfit section affect the global productId?

I thought I read that the only interactions for Your Outfit should be the ability to add or remove items from the outfit list.

**Review by addisonhernandez — CHANGES_REQUESTED** [review]
Can you take another look at lines 32 & 33 of YourOutfit.jsx and confirm with the BRD that clicking the images should change the overall selected product?

File: client/src/components/RelatedItems/YourOutfit.jsx (hunk 3 above)
Comment (Acid-Override) [reply client/src/components/RelatedItems/YourOutfit.jsx:32]: I know...I sort of left it as a place holder and was enjoying the click functionality!

**Review by addisonhernandez — APPROVED** [review]

**Comment by Acid-Override** [comment]
It does seem strange that one group of photos are clickable but not the other group. That seems like a poor user experience but I will update this tomorrow after we discuss. Thanks for the feed back!

**Pull request merged**